packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -140,16 +191,28 @@ async function* saveOps(
     // Split into 100K ops
     const chunk = 100 * 1000;
 
+    let sequencedMessages: ISequencedDocumentMessage[] = [];
+    // current sequence number to read from
+    let curr = 1;
     // Figure out first file we want to write to
     let index = 0;
     if (files.length !== 0) {
-        index = (files.length - 1);
+        index = files.length - 1;
+        const name = filenameFromIndex(index);
+        const fileContent = fs.readFileSync(`${dir}/messages${name}.json`, { encoding: "utf-8" });
+        const messages: ISequencedDocumentMessage[] = JSON.parse(fileContent);
+        // get current sequence number from the last op read, and plus one
+        curr = messages[messages.length - 1].sequenceNumber + 1;
     }
-    let curr = index * chunk + 1;
 
-    let sequencedMessages: ISequencedDocumentMessage[] = [];
     while (true) {
         const result: IteratorResult<ISequencedDocumentMessage[]> = await gen.next();
+        // if there was gap in ops, change curr to currSeq,
+        // and index should be 0 anyways because we ask user to delete files
+        curr = seqNumMismatch ? index * chunk + currSeq : curr;
```

Comment:
I am not able to understand why you are adding currseq to index * chunk. Lets say we had 4 files storing ops till 350K. Then you fetched op 350001 and got error and firstAvailableDelta was 380k and then you will set seqNumMismatch to true. But then here you will set curr = 300K + currSeq(380k) = 680K.

Reply:
and if there were no files and you have firstAvailableDelta as 380K, then you can just set curr to currseq.

Reply:
right, if  seqNumMismatch is true, index will be 0

Reply:
I don't think so it is always true.
`    try {
        await teststream.read();
    } catch (error) {
        statusCode = error.getTelemetryProperties().statusCode;
        innerMostErrorCode = error.getTelemetryProperties().innerMostErrorCode;
        // if there is gap between ops, catch the error and check it is the error we need
        if (statusCode !== 410 || innerMostErrorCode !== "fluidDeltaDataNotAvailable") {
            throw error;
        }
        // get firstAvailableDelta from the error response, and set current sequence number to that
        response = JSON.parse(error.getTelemetryProperties().response);
        currSeq = response.error.firstAvailableDelta;
        lastSeq = currSeq - 1;
        seqNumMismatch = true;
    }`

In this we set it to true in catch part but index does not need to be 0 here.

Reply:
Also if you believe that index will be 0 in case seqNumMismatch is true, then why are you multiplying it with chunk?

Reply:
the try catch block is after reading from existing files, if we reached there, then it means there is no existing files, so index is 0

Reply:
" if we reached there, then it means there is no existing files," I don't think this statement is true. We reach there even if there are files.